src/wallet/rpc/transactions.cpp

```diff
@@ -135,8 +135,12 @@ static UniValue ListReceived(const CWallet& wallet, const UniValue& params, cons
         if (is_change) return; // no change addresses
 
         auto it = mapTally.find(address);
-        if (it == mapTally.end() && !fIncludeEmpty)
+        if (it == mapTally.end() && !fIncludeEmpty) {
```

Comment:
In https://github.com/bitcoin/bitcoin/commit/a1d90fb3b1baefeffb59cf00bad4be3268040ce5 "wallet: 'Filter' out 'send' addresses from 'listreceivedby*'"

For making it concise.

```diff
--- a/src/wallet/rpc/transactions.cpp
+++ b/src/wallet/rpc/transactions.cpp
@@ -73,6 +73,8 @@ struct tallyitem
         auto it = mapTally.find(address);
-        if (it == mapTally.end() && !fIncludeEmpty) {
-            return;
-        } else if (it == mapTally.end()) {
-            LOCK(wallet.cs_wallet);
+        if (it == mapTally.end()) {
+            if (!fIncludeEmpty) return;
             if (!wallet.IsMine(address)) return; // no send addresses
         }
```